nwbwidgets/panel.py

```diff
@@ -127,18 +131,26 @@ def create_components_dandi_source(self, args=None):
             description="File:",
             layout=widgets.Layout(width="400px", overflow=None),
         )
-        self.cache_checkbox = widgets.Checkbox(
-            description="cache",
-        )
-        self.cache_path_text = widgets.Text("nwb-cache")
-        self.cache_path_text.layout.visibility = "hidden"
-        self.source_dandi_file_button = widgets.Button(icon="check", description="Load file")
 
+        if self.enable_cache:
+            self.cache_checkbox = widgets.Checkbox(
+                description="cache",
+            )
+            self.cache_checkbox.observe(self.toggle_cache)
+            self.cache_path_text = widgets.Text("nwb-cache")
+            self.cache_path_text.layout.visibility = "hidden"
+            self.cache_row = widgets.HBox([self.cache_checkbox, self.cache_path_text])
+        else:
+            self.cache_checkbox = None
+            self.cache_path_text = None
+            self.cache_row = widgets.HBox([])
+
+        self.source_dandi_file_button = widgets.Button(icon="check", description="Load file")
         self.source_dandi_vbox = widgets.VBox(
             children=[
                 self.source_dandi_id,
                 self.source_dandi_file_dropdown,
-                widgets.HBox([self.cache_checkbox, self.cache_path_text]),
+                self.cache_row,
```

Comment:
I'd prefer to just not include the cache row here, instead of making it an empty HBox.